### ISSUE TITLE
Delete RoadName data when road segment is deleted

### ISSUE THREAD

**Issue by boformer** [PropaneDragon/RoadNamer#9]
It seems like your mod does not do that.

**Comment by PropaneDragon** [comment]
Ah yeah, this is on my to-do list. Unfortunately I'm unable to work on it until after the new year due to me not being at home. Thanks.

**Comment by boformer** [comment]
I am working on the same feature for my mod. It will probably involve using detours.

Maybe you can use it as well.

**Comment by PropaneDragon** [comment]
That'd be cool, thanks. I can't really think of a way of doing it without detours, unless there's some callback I can use.

**Comment by boformer** [comment]
I am now using this code snippet (Detouring `NetManager.ReleaseSegment`):

``` c#
public new void ReleaseSegment(ushort segment, bool keepNodes)
{
    string caller = new System.Diagnostics.StackFrame(1).GetMethod().Name;

    switch (caller)
    {
        case "MoveMiddleNode": // segment that was modified because user added network, keep data until replacement segments were created
            MoveMiddleNode_releasedSegment = segment; // save segment id
            break;
        case "SplitSegment": // segment that was split by new node, keep data until replacement segments were created
            SplitSegment_releasedSegment = segment; // save segment id
            break;
        case "DeleteSegmentImpl": // segment deleted with bulldozer by user, delete data
        default: // unknown caller, delete data
            NetEventManager.instance.PostReleaseEvent(segment);
            break;
    }

    // Call original method
    RedirectionHelper.RevertRedirect(_NetManager_ReleaseSegment_original, _NetManager_ReleaseSegment_state);
    NetManager.instance.ReleaseSegment(segment, keepNodes);
    RedirectionHelper.RedirectCalls(_NetManager_ReleaseSegment_original, _NetManager_ReleaseSegment_detour);
}
```

I don't know what happens when 2 mods detour the same method. I guess it works as long as you call the original method, and if the detours are deployed and reverted in the right order, like this:
1. Mod-A Deploy
2. Mod-B Deploy
3. Mod-B Revert
4. Mod-A Revert

It would be great to have a shared "hook" mod or library that makes this process safer.

**Comment by tony56a** [comment]
Hate to post something on a super-old issue, but I just wanted to mention that putting in a ThreadingExtensionBase and monitoring relevant net segments, like in Skylines-Overwatch, seems to yield some success( and I'm guessing less likely to potentially break with updates/overlap with other mods then the detour method ), something like:

``` C#
public override void OnAfterSimulationTick()
{
    NetSegment[] buffer = mNetManager.m_segments.m_buffer;
    List<ushort> segments = new List<ushort>(RouteManager.Instance().m_routeDict.Keys);
    foreach (ushort segment in segments) {
        if( (buffer[segment].m_flags) == NetSegment.Flags.None) {
            RouteManager.Instance().DelRoadRoute(segment);
        }
    }
    base.OnAfterSimulationTick();
}
```

**Comment by tony56a** [comment]
Fixed with #12
